.github/workflows/ci.yml

```diff
@@ -1,17 +1,77 @@
 name: CI
-
-on: [push]
-
+on: [push, pull_request]
 jobs:
-  build:
-
+  Scalafmt:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: olafurpg/setup-scala@v2
+      - name: Formatting
+        run: ./bin/scalafmt --test
+  Scalafix:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: olafurpg/setup-scala@v2
+      - name: Linting
+        run: csbt scalafixCheck
+  Docusaurus:
     runs-on: ubuntu-latest
-    
     steps:
-    - uses: actions/checkout@v1
-    - name: Run a one-line script
-      run: echo Hello, world!
-    - name: Run a multi-line script
-      run: |
-        echo Add other actions to build,
-        echo test, and deploy your project.
+      - uses: actions/checkout@v1
+      - uses: olafurpg/setup-scala@v2
+      - name: Website
+        run: csbt docs/docusaurusCreateSite
+  Build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        tool: [sbt, mill, gradle, maven, feature]
+    steps:
+      - uses: actions/checkout@v1
+      - uses: olafurpg/setup-scala@v2
+      - name: Run command
+        run: csbt 'slow/testOnly -- tests.${{ matrix.tool }}'
+  PresentationCompiler:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        scala:
+          - "2.13.0"
```

Comment:
Also, `2.13.11

Reply:
We'll need to add a sbt command to auto-generate this matrix to ensure it's in sync with the build. I don't expect to merge this PR for now because GitHub Actions don't support re-running individual checks yet. I'm just trying to get a feel for how much we can push it 😄

Reply:
We will also need to ensure that the `unit/test` cross-matrix covers all the packages under `tests`

Reply:
See also https://twitter.com/milessabin/status/1176058458262704128